pkg/backend/httpstate/client/api.go

```diff
@@ -141,6 +141,11 @@ func pulumiAPICall(ctx context.Context, d diag.Sink, cloudAPI, method, path stri
 		if err := writer.Flush(); err != nil {
 			return "", nil, errors.Wrapf(err, "flushing compressed payload")
 		}
+		// gzip.Writer will not actually write everything unless it is closed.
+		// without this, the compressed bytes do not decompress properly e.g. in python.
+		if err := writer.Close(); err != nil {
+			return "", nil, errors.Wrapf(err, "closing compressed payload")
+		}
```

Comment:
I believe it is the case that the Flush above is no longer needed, as the Close will automatically flush?

Reply:
Yes, after checking the [documentation for gzip.NewWriter](Close closes the Writer by flushing any unwritten data to the underlying io.Writer and writing the GZIP footer), I confirm that Close() calls Flush(). I have implemented the change.